src/test/java/io/wgj/ctci/StackTest.java

```diff
@@ -0,0 +1,50 @@
+package io.wgj.ctci;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class StackTest {
+    @Test
+    public void testStackOneInt() {
+        Stack s = new Stack();
+        s.push(1);
+        int r = s.pop();
+        assertEquals(1, r);
+    }
+
+    @Test
+    public void testStackMultipleInts() {
+        ArrayList<Integer> expected = new ArrayList<>();
+        for (int i = 2; i < 10; i = i + 2) {
+            expected.add(i);
+        }
+
+        Stack s = new Stack();
+        for (int e : expected) {
+            s.push(e);
+        }
+
+        Collections.reverse(expected);
+        for (int e : expected) {
+            assertEquals(e, s.pop());
+        }
+    }
+
+    @Test(expected = ArrayIndexOutOfBoundsException.class)
+    public void testStackOverFlow() {
+        Stack s = new Stack();
+        for (int i = 0; i < 11; i++) {
```

Comment:
I think this suffers from the problem that if the exception were thrown for one of the first 10 calls for push(), it would still pass.  I think what you want it to make 10 push calls, then make an [@Rule](https://junit.org/junit4/javadoc/4.12/org/junit/rules/ExpectedException.html) that expects ArrayIndexOutOfBoundsException, then make the 11th call that should fail.